### ISSUE TITLE
Diagonal corridors don't work well

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1170]
See this image from the 20160402 build:

![diagonal](https://cloud.githubusercontent.com/assets/13840686/14764424/4b74fef2-09b7-11e6-9eab-66423eda9257.jpg)

This is created by the AI and it looks silly. Much more so when creatures walk through it because they turn 90 degrees on every tile. This situation should be avoided. I see three possible solutions:

1) Don't have the AI dig diagonal lines, just straight lines.
2) Introduce special wall tiles for when diagonal tunnels are fortified
3) Make creatures cut corners so that in this corridor they walk straight like actual people would when they walk such a tunnel.

**Comment by hwoarangmy** [comment]
Well, here, it's not easy to change. When computing path, we assume a creature is on a central tile and goes to the center of the next one. To go diagonally, both tiles that make a square with the diagonal tiles must be dug (otherwise, they would go through half of the wall). Which is never the case in a corridor such as the one on the screenshot.

However, I agree we should try to do something.

**Comment by Bertram25** [comment]
I must say that, as a player, I wouldn't be surprised to see such a behavior as I expect playing against AI.
I agree this looks silly as in clearly non-human behavior. But what would be the simplest solution to this one in your opinion?
I'd advised against adding special cases in the path finding algorithm as it may add unnecessary corner cases in the implementation.

**Comment by Loobinex** [comment]
![walkplease](https://cloud.githubusercontent.com/assets/13840686/14770530/501cabea-0a74-11e6-8427-c211c46c6b0f.jpg)

Currently the creatures follow the red path. They should follow the blue path.
This is actually not a fringe case, but how pathfinding could always work by not assuming to be on the center of the tile but working with sub-tiles.

I don't think the current behavior is a big problem though.

**Comment by Bertram25** [comment]
> Currently the creatures follow the red path. They should follow the blue path.

Ah, I see better what you mean now. IMHO, sub-tiles have nothing to do with this, but rather permitting creatures to reach a tile not forcefully at the center to actually "reach" it is part of the fix.
In other words, the tile offset chosen should be tweaked to offer the shortest path possible, but this is not trivial.

**Comment by Loobinex** [comment]
> but this is not trivial.

Who said it was?

**Comment by Bertram25** [comment]
> Who said it was?

eh eh. :)

**Comment by hwoarangmy** [comment]
Yes, the problem is that it depends on how the corridor is built. As @Bertram25 said, this information is likely to be found at pathfinding and, in most of the cases, it will be useless as, excepted AI, players will not build such corridors. So, adding complexity to pathfinding (which may become a bottleneck) for a specific case is not a good way to go IMHO. However, without having closely looked, I've no easy way to do so.
